### PR TITLE
Deprecate old Page API

### PR DESCRIPTION
**Pull request by teryk**
check the last 2 commits about the Page deprecated API

### CODE REVIEW

File: server/sonar-server/src/main/java/org/sonar/server/ui/DeprecatedViews.java [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  }
+
+  public DeprecatedViews(View[] views) {
+    this.views = views;
+  }
+
+  @Override
+  public void start() {
+    for (View view : views) {
+      LOGGER.warn("View '{}' ({}) is ignored. See {} to define pages.", view.getTitle(), view.getId(), PageDefinition.class.getCanonicalName());
```
Comment (simonbrandhof) [inline server/sonar-server/src/main/java/org/sonar/server/ui/DeprecatedViews.java]: The term "view" is misleading. As the only implementation was the deprecated `Page`, I suggest to use the term "Page" in the warning log.

File: sonar-plugin-api/src/main/java/org/sonar/api/web/RubyRailsPage.java [hunk 2]
```diff
@@ -17,11 +17,14 @@
...
  */
+
 package org.sonar.api.web;
 
 /**
  * @since 1.11
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored. 
  */
+@Deprecated
 public interface RubyRailsPage extends Page {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/RubyRailsPage.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/Page.java [hunk 3]
```diff
@@ -17,11 +17,14 @@
...
  */
+
 package org.sonar.api.web;
 
 /**
  * @since 2.0
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
+@Deprecated
 public interface Page extends View {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/Page.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/DefaultTab.java [hunk 4]
```diff
@@ -26,9 +27,11 @@
 
 /**
  * @since 2.0
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.TYPE)
+@Deprecated
 public @interface DefaultTab {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/DefaultTab.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/View.java [hunk 5]
```diff
@@ -17,16 +17,19 @@
...
 import org.sonar.api.server.ServerSide;
 
 /**
  * @since 1.11
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @ServerSide
 @ExtensionPoint
+@Deprecated
 public interface View {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/View.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/ui/DeprecatedViews.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+import org.sonar.api.utils.log.Loggers;
+import org.sonar.api.web.View;
+import org.sonar.api.web.page.PageDefinition;
+
+/**
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class logs warnings at startup if there are pages using the old page API.
+ */
+@ServerSide
+@Deprecated
+public class DeprecatedViews implements Startable {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ui/DeprecatedViews.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel4.java [hunk 7]
```diff
@@ -230,7 +231,8 @@ protected void configureLevel() {
 
     add(
       PluginDownloader.class,
-      Views.class,
+      DeprecatedViews.class,
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel4.java]: MINOR Remove this use of "DeprecatedViews"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/PageDecoration.java [hunk 8]
```diff
@@ -17,36 +17,19 @@
...
-&lt;% end %&gt;
-</pre>
- *
  * @since 3.3
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @ServerSide
 @ExtensionPoint
+@Deprecated
 public abstract class PageDecoration extends AbstractRubyTemplate {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/PageDecoration.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/ResourceLanguage.java [hunk 9]
```diff
@@ -28,9 +29,11 @@
  * The view is displayed only if the selected resource has the same language.
  *
  * @since 1.11
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.TYPE)
+@Deprecated
 public @interface ResourceLanguage {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/ResourceLanguage.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/Description.java [hunk 10]
```diff
@@ -17,15 +17,20 @@
...
 import java.lang.annotation.RetentionPolicy;
 import java.lang.annotation.Target;
 
+/**
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
+ */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.TYPE)
+@Deprecated
 public @interface Description {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/Description.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/ResourceScope.java [hunk 11]
```diff
@@ -29,9 +30,11 @@
  * The view is displayed only if the selected resource has the same qualifier.
  *
  * @since 1.11
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.TYPE)
+@Deprecated
 public @interface ResourceScope {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/ResourceScope.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: server/sonar-server/src/main/java/org/sonar/server/ui/PageDecorations.java [hunk 12]
```diff
@@ -17,18 +17,20 @@
...
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.web.PageDecoration;
 
 /**
  * @since 3.3
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class has not effect.
  */
 @ServerSide
+@Deprecated
 public final class PageDecorations {
```
Comment (SonarTech) [inline server/sonar-server/src/main/java/org/sonar/server/ui/PageDecorations.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/NavigationSection.java [hunk 13]
```diff
@@ -28,9 +29,11 @@
  * The Page is displayed only in listed sections. This annotation is ignored on Widgets.
  * 
  * @since 1.11
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.TYPE)
+@Deprecated
 public @interface NavigationSection {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/NavigationSection.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/page/PageDefinition.java [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+ *  }
+ * }
+ * </pre>
+ *
+ * @since 6.3
+ */
+
+@ServerSide
+@ExtensionPoint
+public interface PageDefinition {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/page/PageDefinition.java]: CRITICAL Annotate the "PageDefinition" interface with the @FunctionalInterface annotation [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1609)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/AbstractRubyTemplate.java [hunk 15]
```diff
@@ -17,34 +17,22 @@
...
- * Build and deploy the plugin in /extensions/plugins. The file /tmp/sample_dashboard_widget.erb will be reloaded on each request.
- * <br>
- * <br>
- * In production environment, you have to return the classloader path, for example "/org/sonar/myplugin/sample_dashboard_widget.erb".
- *
  * @since 1.11
+ * @deprecated since 6.3. This class is ignored.
  */
+@Deprecated
 public abstract class AbstractRubyTemplate {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/AbstractRubyTemplate.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

File: sonar-plugin-api/src/main/java/org/sonar/api/web/ResourceQualifier.java [hunk 16]
```diff
@@ -28,9 +29,11 @@
  * The view is displayed only if the selected resource has the same qualifier.
  *
  * @since 1.11
+ * @deprecated since 6.3 see {@link org.sonar.api.web.page.PageDefinition}. This class is ignored.
  */
 @Retention(RetentionPolicy.RUNTIME)
 @Target(ElementType.TYPE)
+@Deprecated
 public @interface ResourceQualifier {
```
Comment (SonarTech) [inline sonar-plugin-api/src/main/java/org/sonar/api/web/ResourceQualifier.java]: INFO Do not forget to remove this deprecated code someday. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1133)

**Comment by SonarTech** [comment]
SonarQube analysis reported 16 issues
* CRITICAL 1 critical
* MINOR 2 minor
* INFO 13 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [PlatformLevel4.java#L239](https://github.com/SonarSource/sonarqube/blob/4fad7bf4138ee75c6e90ab3246e29498e1e6fac6/server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel4.java#L239): Remove this use of "PageDecorations"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

File: server/sonar-server/src/main/java/org/sonar/server/ui/DeprecatedViews.java (hunk 1 above)
Comment (teryk) [inline server/sonar-server/src/main/java/org/sonar/server/ui/DeprecatedViews.java]: Actually there was `Page` and `Widget`, but I can have a smarter message that makes the difference

**Comment by SonarTech** [comment]
SonarQube analysis reported 15 issues
* MINOR 2 minor
* INFO 13 info

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [PlatformLevel4.java#L239](https://github.com/SonarSource/sonarqube/blob/874c52d475d49e91f29e0d4faa10cfb2b8926f70/server/sonar-server/src/main/java/org/sonar/server/platform/platformlevel/PlatformLevel4.java#L239): Remove this use of "PageDecorations"; it is deprecated. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)

**Pull request closed without merging**